### PR TITLE
Add sizing to menu divider

### PR DESCRIPTION
**Pull request by AndreiBaicu26**
Add the sizing attribute to the sp-menu-divider component. 

## Related issue(s)
[HZ-20189](https://jira.corp.adobe.com/browse/HZ-20189)

-

## Motivation and context

Allow developers to align the sp-menu-divider wih sp-divider sizing.
## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Westbrook** [comment]
@AndreiBaicu26 great start here! If you could add some stories to support the addition of this new option and follow the instructions in https://github.com/adobe/spectrum-web-components#keeping-ci-assets-updated to update the VRTs tracking that output, we should be good to go.

If you have any trouble getting that to go through, I just onboarded @monkeyjabs to the process and I'm sure he'd love a chance to test out his new knowledge here.

**Comment by AndreiBaicu26** [comment]
> @AndreiBaicu26 great start here! If you could add some stories to support the addition of this new option and follow the instructions in https://github.com/adobe/spectrum-web-components#keeping-ci-assets-updated to update the VRTs tracking that output, we should be good to go.
> 
> If you have any trouble getting that to go through, I just onboarded @monkeyjabs to the process and I'm sure he'd love a chance to test out his new knowledge here.

Thanks for the info, will get into it and if I get stuck I'll ping @monkeyjabs. 🙌🏻

File: packages/menu/stories/menu-divider.stories.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import '@spectrum-web-components/menu/sp-menu-divider.js';
+import '@spectrum-web-components/menu/sp-menu-item.js';
+import '@spectrum-web-components/popover/sp-popover.js';
+
+export default {
+    component: 'sp-menu-divider',
+    title: 'Menu Divider',
+};
+
+export const large = (): TemplateResult => {
```
Comment (Westbrook) [inline packages/menu/stories/menu-divider.stories.ts:23]:
```suggestion
export const s = (): TemplateResult => {
```

File: packages/menu/stories/menu-divider.stories.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                <sp-menu-divider size="s"></sp-menu-divider>
+                <sp-menu-item>Save Selection</sp-menu-item>
+                <sp-menu-divider size="s"></sp-menu-divider>
+                <sp-menu-item>Create group</sp-menu-item>
+            </sp-menu>
+        </sp-popover>
+    `;
+};
+
+export const medium = (): TemplateResult => {
```
Comment (Westbrook) [inline packages/menu/stories/menu-divider.stories.ts:42]:
```suggestion
export const m = (): TemplateResult => {
```

**Review by Westbrook — COMMENTED** [review]
Looking nice. There's not a lot to the template, but if you were amenable, it would be great to leverage a single template and pass the size into it, just in case we need/want to make alterations later, that way it can be all in one place. See the [stories for Avatar](https://github.com/adobe/spectrum-web-components/blob/main/packages/avatar/stories/avatar.stories.ts) as an example of how that works.

**Comment by AndreiBaicu26** [comment]
> Looking nice. There's not a lot to the template, but if you were amenable, it would be great to leverage a single template and pass the size into it, just in case we need/want to make alterations later, that way it can be all in one place. See the [stories for Avatar](https://github.com/adobe/spectrum-web-components/blob/main/packages/avatar/stories/avatar.stories.ts) as an example of how that works.

Thanks for the input , I updated the stories so we can pass a paramater of sizing to visualize the different types of the `menu-divider` :D 🙌🏻

**Pull request merged**